org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardPageObjectsSelectionForPull.java

```diff
@@ -222,81 +195,190 @@ public Object[] getElements(Object inputElement) {
 		@Override
 		public Object[] getChildren(Object parentElement) {
 			if (parentElement instanceof IRepositoryModifiedObjects) {
-				IRepositoryModifiedObjects modifiedObjectsForRepo = (IRepositoryModifiedObjects) parentElement;
-				return modifiedObjectsForRepo.getModifiedObjects().toArray();
-			}
-			return null;
-		}
+				IRepositoryModifiedObjects repo = (IRepositoryModifiedObjects) parentElement;
+				List<IOverwriteObject> all = repo.getModifiedObjects();
```

Comment:
Give a better name for the variable. Example: modifiedObjects.

Reply:
Then let's keep the name of the IRepositoryModifiedObjects variable as repo itself.

For the List of IOverwriteObjects, you can name is as modifiedObjects or modifiedObjectsForRepository.
Also, may be we need a null check for this list then, because we use subList method from the array list.